scripts/o3de/o3de/repo.py

```diff
@@ -422,3 +485,81 @@ def search_o3de_object(manifest_json, o3de_object_uris, search_func):
                     if result_json_data:
                         return result_json_data
     return None
+
+def set_repo_enabled(repo_uri:str, enabled:bool) -> int:
+    repo_uri = f'{repo_uri}/repo.json'
+
+    # avoid downloading if the file already exists and is valid
+    repo_json_cache_file, _ = get_cache_file_uri(repo_uri)
+    repo_json_data = manifest.get_json_data_file(repo_json_cache_file, "repo", validation.valid_o3de_repo_json)
+    if not repo_json_data:
+        # attempt to download the repo.json 
+        repo_json_cache_file = download_repo_manifest(repo_uri)
+        if not repo_json_cache_file.is_file():
+            logger.error(f'{repo_json_cache_file} could not be downloaded')
+            return 1
+
+        repo_json_data = manifest.get_json_data_file(repo_json_cache_file, "repo", validation.valid_o3de_repo_json)
+        if not repo_json_data:
+            logger.error(f'Repository JSON {repo_json_cache_file} could not be loaded or is missing required values')
+            repo_json_cache_file.unlink()
+            return 1
+
+    with repo_json_cache_file.open('w') as f:
+        try:
+            time_now = datetime.now()
+            # %H is 24 hour
+            time_str = time_now.strftime('%d/%m/%Y %H:%M')
```

Comment:
repeat: I recommend using the ISO8601 format for the [date](https://wikiless.org/wiki/ISO_8601?lang=en#Dates) part of the datetime at least.